Docs for request logging (#12363)
apache/druid: docs/configuration/index.md

2019-01-14T10:00:00.000Z        127.0.0.1       {"sqlQuery/time":100,"sqlQuery/bytes":600,"success":true,"identity":"user1"}  {"query":"SELECT page, COUNT(*) AS Edits FROM wikiticker WHERE __time BETWEEN TIMESTAMP '2015-09-12 00:00:00' AND TIMESTAMP '2015-09-13 00:00:00' GROUP BY page ORDER BY Edits DESC LIMIT 10","context":{"sqlQueryId":"c9d035a0-5ffd-4a79-a865-3ffdadbb5fdd","nativeQueryIds":"[490978e4-f5c7-4cf6-b174-346e63cf8863]"}}
```

#### Emitter request logging

> The `emitter` request logger emits every request to some external location, which is configured through an [emitter](#enabling-metrics).

Reviewer: ```suggestion
The `emitter` request logger emits every request to the external location specified in the [emitter](#enabling-metrics) configuration.
```

|Property|Description|Default|
|--------|-----------|-------|
|`druid.request.logging.feed`|Feed name for requests.|none|

#### SLF4J request logging

> The `slf4j` request logger logs every request via  SLF4J. Native queries are serialized into JSON in the log message regardless of the SLF4J format specification. Requests are logged under the class `org.apache.druid.server.log.LoggingRequestLogger`.

Reviewer: ```suggestion
The `slf4j` request logger logs every request using SLF4J. It serializes native queries into JSON in the log message regardless of the SLF4J format specification. Requests are logged under the class `org.apache.druid.server.log.LoggingRequestLogger`.
```

|Property|Description|Default|
|--------|-----------|-------|

> |`druid.request.logging.setMDC`|If MDC entries should be set in the log entry. Your logging setup still has to be configured to handle MDC to format this data.|false|

Reviewer: ```suggestion
|`druid.request.logging.setMDC`|If you want to set MDC entries within the log entry, set this value to "true"  to handle the MDC data format.|false|
```
not  !00% sure  what this was trying to say, verify before accepting change.

Author: ```suggestion
|`druid.request.logging.setMDC`|If you want to set MDC entries within the log entry, set this value to `true`. Your logging system must be configured to support MDC to format this data.|false|
```

> |`druid.request.logging.setContextMDC`|If the Druid query `context` should be added to the MDC entries. Has no effect unless `setMDC` is `true`.|false|

Reviewer: ```suggestion
|`druid.request.logging.setContextMDC`|Set to "true" to add  the Druid query `context` to the MDC entries. Only applies when `setMDC` is `true`.|false|
```

|`resultOrdering`|The ordering of results|
|`descending`|If the query is a descending query|

#### Filtered request logging

The `filtered` request logger filters requests based on the query type or how long a query takes to complete.

> For native queries, requests are only logged when the `query/time` metric is above the user-provided threshold.

Reviewer: ```suggestion
For native queries, the logger only logs requests when the `query/time` metric exceeds the user-provided threshold.
```

Author: ```suggestion
For native queries, the logger only logs requests when the `query/time` metric exceeds the threshold provided in `queryTimeThresholdMs`.
```

> For SQL queries, requests are only logged when the `sqlQuery/time` metric is above the user-provided threshold.

Reviewer: ```suggestion
For SQL queries, it only logs requests when the `sqlQuery/time` metric exceeds the user-provided threshold.
```
for line 317 too,  is this a specific threshold we can mention. For example when sqlQuery/time exceeds the threshold set in <config>

Author: ```suggestion
For SQL queries, it only logs requests when the `sqlQuery/time` metric exceeds threshold provided in `sqlQueryTimeThresholdMs`.
```

##### Graphite Emitter

> To use graphite as emitter set `druid.emitter=graphite`. For configuration details, see the [documentation](../development/extensions-contrib/graphite.md) for the Graphite emitter Druid extension.

Reviewer: ```suggestion
To use graphite as emitter set `druid.emitter=graphite`. For configuration details, see [Graphite emitter](../development/extensions-contrib/graphite.md) for the Graphite emitter Druid extension.
```
